src/VisualStudio/Core/Def/Microsoft.VisualStudio.LanguageServices.csproj

```diff
@@ -163,6 +164,17 @@
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
     <PackageReference Include="System.Threading.Tasks.Dataflow" Version="$(SystemThreadingTasksDataflowVersion)" />
     <PackageReference Include="Nerdbank.Streams" Version="$(NerdbankStreamsVersion)" />
+    <!-- 
+        This is necessary because Microsoft.VisualStudio.LanguageServer.Client.Implementation has transitive dependencies on Microsoft.VisualStudio.Workspace.
+        This package causes Namespace/Type collisions with Microsoft.CodeAnalysis.Workspace. We only reference ILanguageServiceBroker2 from here so the simplest
+        fix is to reference the Microsoft.VisualStudio.LanguageServer.Client.Implementation dll directly.
+        
+        The fix for this is tracked internally by https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1542016
+    -->
+    <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client.Implementation" Version="$(MicrosoftVisualStudioLanguageServerClientImplementation)" ExcludeAssets="all" GeneratePathProperty="true" />
+    <Reference Include="$(PkgMicrosoft_VisualStudio_LanguageServer_Client_Implementation)\lib\net472\Microsoft.VisualStudio.LanguageServer.Client.Implementation.dll">
+      <Private>true</Private>
+    </Reference>
```

Comment:
@dibarbet is the intent that VS.LS takes a direct dep on LSP here?  Or would it be better to hide this behind a language service, and the lang service decides to defer to LSP?

Reply:
We publish the Microsoft.VisualStudio.LanguageServices package for other developers to create VS extension built on top of Roslyn. It does not appear that the Microsoft.VisualStudio.LanguageServer.Client.Implementation package is published to NuGet. This creates a problem for acquisition.

Reply:
Yes, that works for us because we add the vssdk feed to our nuget.config and those packages are published following a release. We do not want to require developers who depend on our packages to add feeds other than nuget.org. It would be best to move this dependency to the implementation project.

Reply:
Ah got it- I've updated the request here for the contracts assembly to be available on nuget.org as well
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1542016/

Reply:
Yeah @JoeRobich suggestion here makes sense - before we merge to main, either
1.  the editor adds a contracts assembly which is public
2.  we move this reference to Microsoft.VisualStudio.LanguageServices.Implementation and add an interface abstraction over it so that we can call it from here.

Reply:
Done